--- coreutils-5.2.1/src/uname.c.orig	2004-01-22 07:27:02.000000000 +0900
+++ coreutils-5.2.1/src/uname.c	2004-06-15 14:30:31.000000000 +0900
@@ -260,7 +260,11 @@
 	    element = processor;
 	}
 #endif
+#ifndef PROCESSOR_TYPE
       print_element (element);
+#else
+      print_element (PROCESSOR_TYPE);
+#endif
     }
 
   if (toprint & PRINT_HARDWARE_PLATFORM)
@@ -284,11 +288,15 @@
 	    element = hardware_platform;
 	}
 #endif
+#ifndef HARDWARE_PLATFORM_TYPE
       print_element (element);
+#else
+      print_element (HARDWARE_PLATFORM_TYPE);
+#endif
     }
 
   if (toprint & PRINT_OPERATING_SYSTEM)
-    print_element (HOST_OPERATING_SYSTEM);
+    print_element (USERLAND);
 
   putchar ('\n');
 
