diff -Naru linux-2.4.20.orig/include/asm-ppc/param.h linux-2.4.20/include/asm-ppc/param.h
--- linux-2.4.20.orig/include/asm-ppc/param.h	2002-12-11 11:39:54.000000000 +0900
+++ linux-2.4.20/include/asm-ppc/param.h	2004-11-12 11:54:01.000000000 +0900
@@ -5,8 +5,12 @@
 #define _ASM_PPC_PARAM_H
 
 #ifndef HZ
+#if defined(CONFIG_OBS200) || defined(CONFIG_OBS50)
+#define HZ 50
+#else
 #define HZ 100
 #endif
+#endif
 
 #define EXEC_PAGESIZE	4096
 
