diff -Naru glibc-2.2.5.orig/sysdeps/unix/sysv/linux/errlist.c glibc-2.2.5/sysdeps/unix/sysv/linux/errlist.c
--- glibc-2.2.5.orig/sysdeps/unix/sysv/linux/errlist.c	2001-07-06 13:56:12.000000000 +0900
+++ glibc-2.2.5/sysdeps/unix/sysv/linux/errlist.c	2002-11-03 18:30:15.000000000 +0900
@@ -37,8 +37,8 @@
 
 const int __old_sys_nerr = OLD_ERRLIST_SIZE;
 
-strong_alias (__old_sys_nerr, _old_sys_nerr);
 weak_alias (__old_sys_nerr, _old_sys_nerr)
+strong_alias (__old_sys_nerr, _old_sys_nerr);
 compat_symbol (libc, __old_sys_nerr, _sys_nerr, GLIBC_2_0);
 compat_symbol (libc, _old_sys_nerr, sys_nerr, GLIBC_2_0);
 weak_alias (__old_sys_errlist, _old_sys_errlist);
